### ISSUE TITLE
require submodules

### ISSUE THREAD

**Issue by antxxxx** [brandonscript/usergrid-nodejs#28]
You mention in the known issues that there is no clean way to require submodules without referencing the full path.
This code works so could the readme be amended to include it
```
var usergridModulePath = require('path').dirname(require.resolve('usergrid'));
var UserGridEntity = require(usergridModulePath + '/lib/entity')
var UsergridQuery = require(usergridModulePath + '/lib/query')
```

**Comment by brandonscript** [comment]
Agreed this is the only working way to do it for now. Not elegant. Feel free to add a PR here and I'll submit one to the origin Apache project.

**Comment by jonasvr** [comment]
when calling /lib/query i get this:

 return (options.query instanceof UsergridQuery) ? {
                              ^
TypeError: Right-hand side of 'instanceof' is not callable

did I do something wrong?

**Comment by brandonscript** [comment]
@jonasvr can you share the actual code you're using? And Node version?

**Comment by jonasvr** [comment]
I'm using node version: v6.11.0

code:
var express = require('express');
var app = express();
var Usergrid = require('usergrid');
var usergridModulePath = require('path').dirname(require.resolve('usergrid'));
var UsergridQuery = require(usergridModulePath + '/lib/query');

**Comment by brandonscript** [comment]
Eh, I hate to use a cliché, but it works for me. I just installed 6.11 via NVM and tested it. Maybe try in a new project?

**Comment by jonasvr** [comment]
in a new project it worked, but started puzzeling:

var express = require('express');
var app = express();
var usergridModulePath = require('path').dirname(require.resolve('usergrid'));
var Usergrid = require('usergrid');

Usergrid.init({
    "appId": "training",
    "orgId": "jsonvr",
    "baseUrl": "https://apibaas-trial.apigee.net/",
    "clientId": "YXA6UUOlMFJtEeeSlRLB41Y48A",
    "clientSecret": "YXA6N5LGfBeOiqTZaovRv_b8Wk23MmM"
});

var UsergridQuery = require(usergridModulePath + '/lib/query');


If I use it after i initiated Usergrid it seems to work.

**Comment by brandonscript** [comment]
Ah, that makes sense. There are initialization params that query depends on that probably don't load correctly. One day I will figure out a proper way of getting these submodules to load into the project automatically.

**Comment by brandonscript** [comment]
Have a look at https://github.com/brandonscript/usergrid-nodejs/tree/submodule-references-%2328

See if this change works for you. Would allow you to do:

    var UsergridQuery = Usergrid.UsergridQuery

(Or just reference `Usergrid.UsergridQuery` directly).
